src/nvim/drawline.c

```diff
@@ -448,6 +451,7 @@ void fill_foldcolumn(win_T *wp, foldinfo_T foldinfo, linenr_T lnum, int attr, in
 
     if (out_buffer) {
       out_buffer[i] = symbol;
+      out_vcol[i] = i >= level ? -1 : (i == closedcol - 1 && closed) ? -2 : -3;
```

Comment:
@bfredl me touchy your code; `fill_foldcolumn()` now fills an array to pass on to `draw_col_buf()` so that it can set the magic vcol values for open/closing a fold when the `%C` foldcolumn item is clicked in the statuscolumn.